app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,46 @@
 class PostsController < ApplicationController
+  helper_method :current_user
+
   def index
-    @user = User.find(params[:id])
+    @user = User.find(params[:user_id])
     @posts = @user.posts.includes(comments: :author)
   end
 
   def show
     @post = Post.includes(comments: :author).find(params[:id])
-    @user = @post.author
+    @user = current_user
+    @new_like = Like.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post successfully created'
+    else
+      redirect_to new_user_post_path(@user), alert: 'Failed to create post'
+    end
+  end
+
+  def create_like
+    @post = Post.find(params[:id])
+    @new_like = @post.likes.build(author: current_user)
+
+    if @new_like.save
+      redirect_to request.referrer, notice: 'Post liked'
+    else
+      redirect_to request.referrer, alert: 'Failed to like post'
+    end
+  end
```

Comment:
:red_circle:  While this might work I recommend that you create a likes controller and move the create_like method inside the posts controller in the likes controllers. This will make your code more organized, testable, and reusable.

Here are the steps on how to create a likes controller:

- Create a new file called `likes_controller.rb` in your` app/controllers` directory.
- Add the following code to the file:

```
class LikesController < ApplicationController
  def create
    like = Post.find(params[:post_id]).likes.new(author: current_user)
    if like.save
      flash[:success] = 'Post liked'
    else
      # error message
      flash.now[:error] = 'Error: Failed to like post'
    end
    redirect_back(fallback_location: users_path)
  end
end
```